### ISSUE TITLE
"Trimap has 102 colors" on a 3 colors image 

### ISSUE THREAD

**Issue by xamxixixo** [pymatting/pymatting#14]
Hello, I painted a trimap in Photoshop with colors picked from lamur_trimap.png. Then I used the [cutout] function. And there was this error appeared:

```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-28-ce857cc21222> in <module>()
      7     "data/nik/nik_trimap.png",
      8     # output cutout path
----> 9     "nik_cutout.png")

3 frames
/content/drive/My Drive/Colab/pymatting/pymatting/cutout/cutout.py in cutout(image_path, trimap_path, cutout_path)
     28         raise ValueError("Input image and trimap must have same size")
     29 
---> 30     alpha = estimate_alpha_cf(image, trimap)
     31 
     32     foreground = estimate_foreground_ml(image, alpha)

/content/drive/My Drive/Colab/pymatting/pymatting/alpha/estimate_alpha_cf.py in estimate_alpha_cf(image, trimap, preconditioner, laplacian_kwargs, cg_kwargs)
     44         preconditioner = ichol
     45 
---> 46     A, b = make_linear_system(cf_laplacian(image, **laplacian_kwargs), trimap)
     47 
     48     x = cg(A, b, M=preconditioner(A), **cg_kwargs)

/content/drive/My Drive/Colab/pymatting/pymatting/laplacian/laplacian.py in make_linear_system(L, trimap, lambda_value, return_c)
     36     h, w = trimap.shape[:2]
     37 
---> 38     is_fg, is_bg, is_known, is_unknown = trimap_split(trimap)
     39 
     40     c = lambda_value * is_known

/content/drive/My Drive/Colab/pymatting/pymatting/util/util.py in trimap_split(trimap, flatten)
    447             "Make sure that your trimaps are stored as PNG instead of JPG.\n"
    448             "If you scaled the trimap, make sure to use nearest filtering:\n"
--> 449             '    load_image("trimap.png", "GRAY", 0.5, "nearest")' % n_colors
    450         )
    451 

ValueError: Trimap has 102 colors, but should have no more than 3 (black, white, gray).
Make sure that your trimaps are stored as PNG instead of JPG.
If you scaled the trimap, make sure to use nearest filtering:
    load_image("trimap.png", "GRAY", 0.5, "nearest")
```

This is the trimap

![nik_trimap](https://user-images.githubusercontent.com/37895275/79205759-3486dd00-7df3-11ea-95ea-568918d58a97.png)

**Comment by 99991** [comment]
When you zoom in you can see that your trimap consists of more than 3 colors (black, white, and several different gray tones):

![zoom](https://user-images.githubusercontent.com/18725165/79206659-f532ad80-7e3f-11ea-98c8-63712a73627f.png)

You can fix your trimap using the following function:

```Python

def fix_trimap(trimap, threshold=0.1):
    is_bg = trimap < threshold
    is_fg = trimap > 1.0 - threshold
    fixed = 0.5 * np.ones_like(trimap)
    fixed[is_fg] = 1
    fixed[is_bg] = 0
    return fixed

trimap = fix_trimap(trimap)
```

**Comment by xamxixixo** [comment]
thank you
